src/main/java/org/simlar/simlarserver/services/subscriberservice/SubscriberService.java

```diff
@@ -124,11 +127,9 @@ public void checkCredentialsWithException(final String simlarId, final String ha
         }
     }
 
-    public int getStatus(final SimlarId simlarId) {
-        if (simlarId == null) {
-            return 0;
-        }
-
-        return subscriberRepository.findHa1ByUsernameAndDomain(simlarId.get(), sharedSettings.domain()).isEmpty() ? 0 : 1;
+    public List<SimlarId> filterSimlarIdsRegistered(final Collection<SimlarId> simlarIds) {
+        final Set<String> usernames = simlarIds.stream().map(SimlarId::get).collect(Collectors.toSet());
+        return subscriberRepository.findUsernameByDomainAndUsernameIn(sharedSettings.domain(), usernames)
+                .stream().map(SimlarId::create).toList();
```

Comment:
Creating batches of ids should help limiting the number of arguments per database call. E.g. https://www.baeldung.com/java-stream-batch-processing#using-collection-api

Reply:
Thank you again @jritzerfeld 
I have refined the implementation according to your suggestion.